lib/compilers/java.js

```diff
@@ -0,0 +1,265 @@
+                return files.filter(f => f.endsWith('.class'));
+            })
+            .then(classFiles => {
+                return Promise.all(classFiles.map(classFile => {
+                    const args = [
+                        // Prints out disassembled code, i.e., the instructions that comprise the Java bytecodes,
+                        // for each of the methods in the class.
+                        '-c',
+                        // Prints out line and local variable tables.
+                        '-l',
+                        classFile
+                    ];
+                    return this.exec(this.compiler.objdumper, args, {maxOutput: maxSize, customCwd: dirPath})
+                        .then(objResult => {
+                            const oneResult = {};
+                            oneResult.asm = objResult.stdout;
+                            if (objResult.code !== 0) {
+                                oneResult.asm = "<No output: javap returned " + objResult.code + ">";
+                            }
+                            return oneResult;
+                        });
+                }));
+            })
+            .then(results => {
+                const merged = {asm: []};
+                for (const result of results) {
+                    const asmBackup = merged.asm;
+                    Object.assign(merged, result);
+                    merged.asm = asmBackup;
+                    merged.asm.push(result.asm);
+                }
+
+                result.asm = merged.asm;
+                return result;
+            });
+    }
+
+
+    optionsForFilter(filters) {
+        // Forcibly enable javap
+        filters.binary = true;
+
+        return [
+            '-Xlint:all',
+        ];
+    }
+
+
+    getArgumentParser() {
+        return argumentParsers.Java;
+    }
+
+    getOutputFilename(dirPath) {
+        return path.join(dirPath, `${path.basename(this.compileFilename, this.lang.extensions[0])}.class`);
+    }
+
+
+    filterUserOptions(userOptions) {
+        const filteredOptions = [];
+        let toSkip = 0;
+
+        const oneArgBlacklist = [
+            // -d directory
+            // Sets the destination directory for class files.
+            "-d",
+            // -s directory
+            // Specifies the directory used to place the generated source files.
+            "-s",
+            // --source-path path or -sourcepath path
+            // Specifies where to find input source files.
+            "--source-path", "-sourcepath"];
+        for (const userOption of userOptions) {
+            if (toSkip > 0) {
+                toSkip--;
+                continue;
+            }
+            if (oneArgBlacklist.indexOf(userOption) !== -1) {
+                toSkip = 1;
+                continue;
+            }
+
+            filteredOptions.push(userOption);
+        }
+
+        return filteredOptions;
+    }
+
+    processAsm(result) {
```

Comment:
Could some tests be added to ensure we don't unintentionally break the this processing function(s)? Otherwise, they look great!

Reply:
+1 to this! Love this PR but having some tests for the asm parsers has been beyond invaluable in ensuring the quality of Compiler Explorer's output.